### ISSUE TITLE
NullPointerException when trying to parse CMAKE_EXPORT_COMPILE_COMMANDS JSON

### ISSUE THREAD

**Issue by jesse-gallagher** [15knots/cmake4eclipse#55]
I've run into a problem trying to make use of the compile_commands.json output from CMake. Namely, it looks like, when cmake4eclipse attempts to find the file after a CDT build, it runs into an NPE trying to build the IPath from the existing project path:

```
java.lang.NullPointerException
	at de.marw.cmake.cdt.language.settings.providers.CompileCommandsJsonParser.tryParseJson(CompileCommandsJsonParser.java:277)
	at de.marw.cmake.cdt.language.settings.providers.CompileCommandsJsonParser.shutdown(CompileCommandsJsonParser.java:663)
	at org.eclipse.cdt.internal.core.ConsoleOutputSniffer.closeConsoleOutputStream(ConsoleOutputSniffer.java:160)
	at org.eclipse.cdt.internal.core.ConsoleOutputSniffer$ConsoleOutputStream.close(ConsoleOutputSniffer.java:68)
	at org.eclipse.cdt.internal.core.ProcessClosure$ReaderThread.close(ProcessClosure.java:98)
	at org.eclipse.cdt.internal.core.ProcessClosure.isAlive(ProcessClosure.java:191)
	at org.eclipse.cdt.core.CommandLauncher.waitAndRead(CommandLauncher.java:273)
	at org.eclipse.cdt.internal.core.BuildRunnerHelper.build(BuildRunnerHelper.java:275)
	at org.eclipse.cdt.make.core.MakeBuilder.invokeMake(MakeBuilder.java:219)
	at org.eclipse.cdt.make.core.MakeBuilder.build(MakeBuilder.java:108)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.resources.Workspace.buildInternal(Workspace.java:487)
	at org.eclipse.core.internal.resources.Workspace.build(Workspace.java:399)
	at org.eclipse.ui.actions.GlobalBuildAction$1.run(GlobalBuildAction.java:177)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
```

This seems likely to be the same line in the current codebase:

https://github.com/15knots/cmake4eclipse/blob/938343d6f794751e97263660893258ffc21124f6/de.marw.cmake/src/main/java/de/marw/cmake/cdt/language/settings/providers/CompileCommandsJsonParser.java#L277

I'm working with cmake4eclipse 1.7.0.201709231854 and this Eclipse environment:

```
eclipse.buildId=4.7.1.M20171009-0410
java.version=1.8.0_111
java.vendor=Oracle Corporation
BootLoader constants: OS=macosx, ARCH=x86_64, WS=cocoa, NL=en_US
Framework arguments:  -product org.eclipse.epp.package.jee.product -product org.eclipse.epp.package.jee.product -keyring /Users/jesse/.eclipse_keyring
Command-line arguments:  -os macosx -ws cocoa -arch x86_64 -product org.eclipse.epp.package.jee.product -data file:/Volumes/Terra/eclipse/darwino-platform/ -product org.eclipse.epp.package.jee.product -keyring /Users/jesse/.eclipse_keyring
```

I'm not sure if there's any more useful contextual information I can provide, but let me know if there is.

**Comment by 15knots** [comment]
Thanks for the stack trace.
The NPE is weird,  `builderCWD` should never be null. Does it happen every time or did it happen just once.
Are the build scripts generated by the plugin or do you use the parser without it.

**Comment by 15knots** [comment]
I could not reproduce this. In my stack trace, there is no `org.eclipse.cdt.make.core.MakeBuilder`, even if I create a `Makefile Project with Existing Code` project,  I always get the `org.eclipse.cdt.managedbuilder.internal.core.CommonBuilder`.
What is the type of your Eclipse project? Do you invoke `make` as an external tool from Eclipse?
Could you post the content of your `.project` file here?

**Comment by jesse-gallagher** [comment]
I’m out of town here for a few days, but I’ll provide more info and the .project file when I’m back later this week.

**Comment by 15knots** [comment]
I found out how you did setup up the project: You ran `cmake -G "Eclipse CDT4 - Unix Makefiles"` to generate the build scripts and then added the `CompileCommandsJsonParser` in the project settings.
Your life would be easier if you simply let the plugin generate the builds scripts, that is main reason why the plugin exists.

Anyway, I can reproduce the NPE now.

**Comment by jesse-gallagher** [comment]
That's it indeed - thanks! I ran into some other oddities with the normal route, but I suspect that those are just due to the weird way I wrote the CMakeLists when converting the project from VS in the first place, so that'll be something for me to sort out.

**Comment by 15knots** [comment]
Fixed in #ca047e75a9446e34993641f7a2a087dca47618eb
